### PR TITLE
Implement functionality to move drones, update their positions and update the grid

### PR DESCRIPTION
**Pull request by Thanoshan**
This PR adds functionality to NEPIADA to actually move the drones given a set of actions. These changes allow for updating the drone's position, and updating the grid with the new drone positions. It also implements the ability to detect boundary collisions as well as drone-on-drone collisions.

Summary of changes in PR:
1. Update initialization to properly accept configuration objects, and update World to store agents as a dictionary with format {ID: Agent}
2. Initialize possible agents with World's Agent dictionary keys to avoid potential errors with duplicate code
3. Add a field to Agent class to store a unique ID, which is numerical to uniquely identify agents
4. Implement move_drones in environment to move the drones on every step
5. Implement move_drone in grid to take an agent and action and do legality checking as well as updating the grid
6. If drones collide, do a second pass in reverse of the subset of drones that failed to see if the position became available, if so move them.

More info to come...

This PR closes issue #22 and makes progress on #19

### CODE REVIEW

File: src/nepiada/utils/agent.py [hunk 1]
```diff
@@ -16,10 +16,10 @@ def __init__(self):
         self.p_vel = None
 
 class Agent(Entity):  # properties of agent entities
-    def __init__(self, type):
+    def __init__(self, type, uid):
```
Comment (pandyah5) [inline src/nepiada/utils/agent.py:19]: This could present a case where two agents might have the same uid. I don't like this method of assigning IDs.

Comment (pandyah5) [reply src/nepiada/utils/agent.py:19]: We should add some checks around it to ensure that uid are unique

File: src/nepiada/utils/config.py [hunk 2]
```diff
@@ -4,3 +4,4 @@ class Config:
     num_adversarial_agents: int = 2
     obs_radius: int = 20
     iterations: int = 100
+    possible_moves : {int : int} = {0 : (0, 0), 1 : (0, 1), 2 : (0, -1), 3 : (-1, 0), 4 : (1, 0)}
```
Comment (pandyah5) [inline src/nepiada/utils/config.py:7]: Add a comment explaining it

File: src/nepiada/utils/grid.py [hunk 3]
```diff
@@ -19,10 +20,50 @@ def __init__(self):
...
-        for agent in agents:
+        for _, agent in agents.items():
             x_coord = agent.p_pos[0]
             y_coord = agent.p_pos[1]
-            self.state[x_coord][y_coord] = agent.type.value
+            # commenting this out for now until we have a better way to render
+            # self.state[x_coord][y_coord] = agent.uid if agent.type == AgentType.TRUTHFUL else -agent.uid
+            self.state[x_coord][y_coord] = agent.uid
+    
+    def move_drone(self, agent, action):
```
Comment (pandyah5) [inline src/nepiada/utils/grid.py:30]: Good implementation!

File: src/nepiada/utils/config.py (hunk 2 above)
Comment (Thanoshan) [reply src/nepiada/utils/config.py:7]: Done.

**Comment by Thanoshan** [comment]
> Good implementation! 

Thanks! :D

File: src/nepiada/utils/agent.py (hunk 1 above)
Comment (Thanoshan) [reply src/nepiada/utils/agent.py:19]: We theoretically could use something like itertools, or just declare a global variable that we increment each time. Though I don't think it's necessary to implement such a check at the class level, I think typically good practice would be to enforce it where the agents are created. What do you think?

Comment (Thanoshan) [reply src/nepiada/utils/agent.py]: Never mind, investigated it and made it global.

**Review by pandyah5 — APPROVED** [review]

**Pull request merged**